### PR TITLE
chore: support min go version 1.18

### PR DESCRIPTION
**Pull request by thinkerou**
and fixes https://github.com/gin-gonic/gin/issues/2665

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ to your code, and then `go [build|run|test]` will automatically fetch the necess
 Otherwise, run the following Go command to install the `gin` package:
 
 ```sh
-$ go get -u github.com/gin-gonic/gin
+$ go install github.com/gin-gonic/gin@latest
```
Comment (mstmdev) [inline README.md:49]: `go install` will not update the `go.mod` file, suggest keeping using `go get -u`.

**Review by mstmdev — CHANGES_REQUESTED** [review]

File: context_1.18_test.go [hunk 2]
```diff
@@ -2,8 +2,8 @@
 // Use of this source code is governed by a MIT style
 // license that can be found in the LICENSE file.
 
-//go:build !go1.17
-// +build !go1.17
+//go:build !go1.19
+// +build !go1.19
```
[the remark this answers is not in the document]
Comment (thinkerou) [reply context_1.18_test.go:6]: thank your remind! I will fix it.
https://github.com/gin-gonic/gin/issues/2665

**Comment by mstmdev** [comment]
Could we delete all the `any.go` files?

**Comment by thinkerou** [comment]
> Could we delete all the `any.go` files?

yes, done. please review, thanks!

**Comment by mstmdev** [comment]
image

Replace all the` interface{}` with `any`.

**Comment by thinkerou** [comment]
> image
> 
> Replace all the `interface{}` with `any`.

@mstmdev thanks a lot! done.

**Review by mstmdev — APPROVED** [review]

**Comment by thinkerou** [comment]
@appleboy please review, thanks!

**Review by appleboy — APPROVED** [review]

**Pull request merged**